### PR TITLE
feat: ability to fetch logs from instances

### LINKED ISSUE #496: Ability to fetch logs from instances

### PR DESCRIPTION
**Pull request by mojtaba-esk**
Closes #496 

## Summary by CodeRabbit

- **New Features**
	- Introduced logging tests to validate functionality for both standalone and sidecar instances.
	- Added methods to retrieve log streams for specific containers in Kubernetes ReplicaSets.
	- Enhanced monitoring capabilities with new methods for retrieving logs based on instance types.

- **Bug Fixes**
	- Improved log retrieval logic to ensure accurate output for different instance configurations.

### CODE REVIEW

File: e2e/basic/logs_test.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+// 	s.Require().NoError(target.Execution().Start(ctx))
+
+// 	// Perform the test
+// 	expectedOutput := "Hello World"
+// 	output, err := target.Execution().ExecuteCommand(ctx, "echo", expectedOutput)
+// 	s.Require().NoError(err)
+
+// 	output = strings.TrimSpace(output)
+// 	s.Assert().Equal(expectedOutput, output)
+// }
```
Comment (smuu) [inline e2e/basic/logs_test.go:29]: Do we need these lines?

File: e2e/basic/logs_test.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	defer logStream.Close()
+
+	logs, err := io.ReadAll(logStream)
+	s.Require().NoError(err)
+
+	logOutput := string(logs)
+	s.Contains(logOutput, expectedLogMsg)
+}
+
+func (s *Suite) TestLogsWithSidecar() {
```
Comment (smuu) [inline e2e/basic/logs_test.go:62]: What is this test for? The `sidecar` instance is never set as a sidecar but started like a standard instance.

**Review by smuu — CHANGES_REQUESTED** [review]
Neat solution 🚀 
Only two small things in the tests.

File: e2e/basic/logs_test.go (hunk 1 above)
Comment (mojtaba-esk) [reply e2e/basic/logs_test.go:29]: oh XI, I forgot to uncomment those lines. they test fetching logs for a pod and the next one test fetching logs for a sidecar.

File: e2e/basic/logs_test.go (hunk 2 above)
Comment (mojtaba-esk) [reply e2e/basic/logs_test.go:62]: thanks for pointing it out. I will fix it.

**Review by smuu — APPROVED** [review]

**Pull request merged**